Homework 17

hcarter3508 opened the pull request:
Completed the assignment with some testing.

hcarter3508 committed:
Update Main.java

hcarter3508 committed:
Homework 17

Implemented encapsulation of specific procedures, tested Koel functionality related to adding a song to a playlist. Also selected an album with only one song in order to verify that indexing a list of one element would work.

Berichicko commented:
Please finish the homework!You got this problem, please correct it. 

![image](https://github.com/Test-Pro-Bootcamp/A47-QA-Automation-Framework/assets/104761482/a03c900f-4d2d-4245-bca8-3b3f9297a6d9)

HINT

**I will try to help you how easy it is to find locators:**

**These steps describe the process of opening the developer console in the Google Chrome browser and making changes to the styles of a popover that appears when hovering over a specific element on a web page.**

For example

 `By.cssSelector("div.success.show")`
 
```
    public String getAddToPlaylistSuccessMsg() {
            WebElement notification = driver.findElement(By.cssSelector("div.success.show"));
            return notification.getText();
        }
```

![image](https://user-images.githubusercontent.com/104761482/235407463-672c5728-57d6-4eb8-ae1b-945c83f12f5f.png)

![image](https://user-images.githubusercontent.com/104761482/235407647-4457ba8f-eba2-4c8f-8f57-bae764eeba43.png)

Browse to the desired page
1.	Open the dev console - F12 on Windows/Linux or option + ⌘ + J on macOS
2.	Select the Sources tab in chrome inspector
3.	In the web browser window, hover over the desired element to initiate the popover
4.	Hit F8 on Windows/Linux (or fn + F8 on macOS or  CTRL+\) while the popover is showing. If you have clicked anywhere on the actual page F8 will do nothing. Your last click needs to be somewhere in the inspector, like the sources tab
5.	Go to the Elements tab in inspector
6.	Find your popover (it will be nested in the trigger element's HTML)
7.	Have fun modifying the CSS
8.	https://www.howtogeek.com/663950/what-your-function-keys-do-in-chrome-devtools/

hcarter3508 commented:
I did not get the same output when I ran it here at the library. In fact, all tests passed. I will test it again now. Thank you!

hcarter3508 commented:
> Task :compileJava
> Task :processResources NO-SOURCE
> Task :classes
> Task :compileTestJava
> Task :processTestResources
> Task :testClasses
SLF4J: No SLF4J providers were found.
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See https://www.slf4j.org/codes.html#noProviders for further details.
Starting ChromeDriver 117.0.5938.92 (67649b10b92bb182fba357831ef7dd6a1baa5648-refs/branch-heads/5938_62@{#14}) on port 59486
Only local connections are allowed.
Please see https://chromedriver.chromium.org/security-considerations for suggestions on keeping ChromeDriver safe.
ChromeDriver was started successfully.
Oct 01, 2023 3:10:50 PM org.openqa.selenium.remote.ProtocolHandshake createSession
INFO: Detected upstream dialect: W3C
Oct 01, 2023 3:10:50 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
WARNING: Unable to find an exact match for CDP version 117, so returning the closest version found: a no-op implementation
Oct 01, 2023 3:10:50 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
INFO: Unable to find CDP implementation matching 117.
Oct 01, 2023 3:10:50 PM org.openqa.selenium.chromium.ChromiumDriver lambda$new$3
WARNING: Unable to find version of CDP to use for . You may need to include a dependency on a specific version of the CDP using something similar to `org.seleniumhq.selenium:selenium-devtools-v86:4.5.0` where the version ("v86") matches the version of the chromium-based browser you're using and the version number of the artifact is the same as Selenium's.
> Task :test
BUILD SUCCESSFUL in 17s
4 actionable tasks: 4 executed
3:10:55 PM: Execution finished ':test --tests "Homework17"'.

Berichicko commented:
> > Task :compileJava
> > Task :processResources NO-SOURCE
> > Task :classes
> > Task :compileTestJava
> > Task :processTestResources
> > Task :testClasses
> > SLF4J: No SLF4J providers were found.
> > SLF4J: Defaulting to no-operation (NOP) logger implementation
> > SLF4J: See https://www.slf4j.org/codes.html#noProviders for further details.
> > Starting ChromeDriver 117.0.5938.92 (67649b10b92bb182fba357831ef7dd6a1baa5648-refs/branch-heads/5938_62@{#14}) on port 59486
> > Only local connections are allowed.
> > Please see https://chromedriver.chromium.org/security-considerations for suggestions on keeping ChromeDriver safe.
> > ChromeDriver was started successfully.
> > Oct 01, 2023 3:10:50 PM org.openqa.selenium.remote.ProtocolHandshake createSession
> > INFO: Detected upstream dialect: W3C
> > Oct 01, 2023 3:10:50 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
> > WARNING: Unable to find an exact match for CDP version 117, so returning the closest version found: a no-op implementation
> > Oct 01, 2023 3:10:50 PM org.openqa.selenium.devtools.CdpVersionFinder findNearestMatch
> > INFO: Unable to find CDP implementation matching 117.
> > Oct 01, 2023 3:10:50 PM org.openqa.selenium.chromium.ChromiumDriver lambda$new$3
> > WARNING: Unable to find version of CDP to use for . You may need to include a dependency on a specific version of the CDP using something similar to `org.seleniumhq.selenium:selenium-devtools-v86:4.5.0` where the version ("v86") matches the version of the chromium-based browser you're using and the version number of the artifact is the same as Selenium's.
> > Task :test
> > BUILD SUCCESSFUL in 17s
> > 4 actionable tasks: 4 executed
> > 3:10:55 PM: Execution finished ':test --tests "Homework17"'.

Hello

Please, re-submit homework for review

hcarter3508 committed:
Update Homework17.java

Added a module to extract a string from a web page using Selenium and a CSS selector.

Berichicko commented:
Hello,

I have checked your homework 17, and I wanted to let you know that you're doing great! Keep up the excellent work!
Best wishes from the HW team (Alex, [aliaksandr@testpro.io](mailto:aliaksandr@testpro.io))!

![804deb64dc406a20a75da05d91a1a69e](https://user-images.githubusercontent.com/104761482/235961479-9afb2841-b657-46c9-99f8-2402f933b843.jpg)
